[BUG] Missing macro definition for FOUR_C_TRILINOS_INTERNAL_VERSION_GE in absence of trilinos commit information.

Dear all,

compiling a more recent version of 4C yields the following preprocessor error
message:

```
/home/foo/builds/4C-fafddbdd2b/src/solver_nonlin_nox/4C_solver_nonlin_nox_group.hpp:93:42: error: missing binary operator before token ‘(’
   93 | #if !(FOUR_C_TRILINOS_INTERNAL_VERSION_GE(2025, 4))
      |                                          ^
/home/foo/builds/4C-fafddbdd2b/src/solver_nonlin_nox/4C_solver_nonlin_nox_group.hpp:189:42: error: missing binary operator before token ‘(’
  189 | #if !(FOUR_C_TRILINOS_INTERNAL_VERSION_GE(2025, 4))
      | 
```

which may point to a missing macro definition.

I've used the 4C commit fafddbdd2b and trilinos 06db4c85065 as specified in the
trilinos install script provided by 4C. 

Digging a bit further it seems that
`cmake/functions/four_c_configure_dependency.cmake` is responsible for emitting
the mentioned macro definition, but only when the git commit hash could be
retrieved from the trilinos installation
(`cmake/configure/configure_Trilinos.cmake`).

And indeed, I compiled trilinos without git metadata, and
`TrilinosRepoVersion.txt` is missing from the Trilinos installation directory.

This appears to be a bug – under these circumstances cmake  should never
configure the project successfully and confront the user with this admittedly
puzzling error message.

I'd be happy to assist you further if more context is needed.

--- a/cmake/configure/configure_Trilinos.cmake
+++ b/cmake/configure/configure_Trilinos.cmake
@@ -46,6 +46,8 @@ if(EXISTS "${Trilinos_DIR}/../../../TrilinosRepoVersion.txt")
   list(GET TrilinosRepoVersionFileLine2 0 _sha)
 
   set(FOUR_C_Trilinos_GIT_HASH ${_sha})
+else()
+  set(FOUR_C_Trilinos_GIT_HASH "unknown")
 endif()
 
 target_link_libraries(

